### PR TITLE
fix(helm): Use correct default hostnames for postgres and redis

### LINKED ISSUE #24218: Helm chart with default values fails to install
The helm chart fails to install, and never gets past the init job.

#### How to reproduce the bug

```sh
helm repo add superset https://apache.github.io/superset
helm repo update
helm upgrade --install my-release superset/superset --namespace default
```

### Expected results

I expected the helm chart to install successfully

### Actual results

```
Release "my-release" does not exist. Installing it now.
Error: failed post-install: timed out waiting for the condition
```

When the `init` job runs, it fails to connect to the postgres DB:

```log
wait-for-postgres 2023/05/25 20:17:20 Problem with dial: dial tcp: lookup my-release-superset-postgresql on 10.43.0.10:53: no such host. Sleeping 1s
wait-for-postgres 2023/05/25 20:17:21 Problem with dial: dial tcp: lookup my-release-superset-postgresql on 10.43.0.10:53: no such host. Sleeping 1s
wait-for-postgres 2023/05/25 20:17:22 Problem with dial: dial tcp: lookup my-release-superset-postgresql on 10.43.0.10:53: no such host. Sleeping 1s
wait-for-postgres 2023/05/25 20:17:23 Problem with dial: dial tcp: lookup my-release-superset-postgresql on 10.43.0.10:53: no such host. Sleeping 1s
```

The Kubernetes services are named `my-release-postgresql` AND NOT `my-release-superset-postgresql`.

#### Screenshots

N/A

### Environment

(please complete the following information):

- browser type and version: N/A
- superset version: `2.1.0` (Helm Chart `0.10.0`)
- python version: N/A
- node.js version: N/A
- any feature flags active: N/A

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

### Additional context

None

### PR DESCRIPTION
**Pull request by NickLarsenNZ**
### SUMMARY
Render the correct hostname for the default postgres and redis services.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

`Secret` resource:

image

`Service` resources:

image

### TESTING INSTRUCTIONS

```sh
helm upgrade --install my-release ~/dev/github.com/apache/superset/helm/superset --namespace default
```

```sh
helm upgrade --install my-release ~/dev/github.com/apache/superset/helm/superset --namespace default --set nameOverride="foo" --set fullnameOverride="bar"
```

### ADDITIONAL INFORMATION

- [x] Has associated issue: Fixes #24218
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by craig-rueda — COMMENTED** [review]
Missing version bump and readme rebuild

**Comment by NickLarsenNZ** [comment]
> Missing version bump and readme rebuild

@craig-rueda this has been done now.

**Review by craig-rueda — APPROVED** [review]

**Comment by craig-rueda** [comment]
Looks like there's still a linter error

File: helm/superset/Chart.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@
 # limitations under the License.
 #
 apiVersion: v2
-appVersion: "2.1.0"
+appVersion: "2.1.1"
```
Comment (NickLarsenNZ) [inline helm/superset/Chart.yaml:18]: @craig-rueda I see I have updated the wrong version. Glad these checks are in place.
I'll fix tomorrow

Comment (NickLarsenNZ) [reply helm/superset/Chart.yaml]: @craig-rueda that's fixed now

**Comment by NickLarsenNZ** [comment]
Good to go now @craig-rueda

**Pull request merged**